### PR TITLE
Added Prims and Kruskal's Algo

### LINKED ISSUE #1908: [Idea]: Radix sort
### Idea Title

Radix Sort

### Idea Description
This algorithm sorts numbers by processing individual digits, starting from the least significant digit to the most significant digit (LSD Radix Sort). It’s suitable for integers or strings.

in sorting algorithms this algorithm is missing
I'd like to add this
@ajay-dhangar can you assign this to me??

### Potential Benefits

- [ ] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [ ] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### LINKED ISSUE #1909: [Idea]:  Shell Sort
### Idea Title

Add a new sorting algorithm

### Idea Description

An optimized version of insertion sort. Shell Sort sorts elements far apart first, then progressively reduces the gap.

@ajay-dhangar can you assign this to me??

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### LINKED ISSUE #1910: [Idea]: Prims Algorithm
### Idea Title

add a new greedy algorithm

### Idea Description

Prim's Algorithm (Minimum Spanning Tree)
is To find a Minimum Spanning Tree (MST) for a weighted, undirected graph.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### LINKED ISSUE #1911: [Idea]: Kruskal's Algorithm
### Idea Title

add a new greedy algorithm

### Idea Description

Another MST algorithm that works by sorting all edges.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by laxmikandivalasa**
### Description
Added Prims and Kruskal's Algorithms

Fixes # 1910 #1911 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X ] My code follows the style guidelines of this project
- [X ] I have performed a self-review of my code
- [ X] I have commented my code, particularly in hard-to-understand areas
- [X ] I have added tests that prove my fix is effective or that my feature works
- [ X] New and existing unit tests pass locally with my changes
- [X ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by laxmikandivalasa** [comment]
@ajay-dhangar can you review this pr

**Comment by laxmikandivalasa** [comment]
@ajay-dhangar can you have a look at these prs

**Comment by laxmikandivalasa** [comment]
can you label this with level 2 since i included 2 issues

File: docs/greedy-algorithms/kruskals-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
+---
+id: kruskals-algorithm  
+title: Kruskal's Algorithm
```
Comment (ajay-dhangar) [inline docs/greedy-algorithms/kruskals-algorithm.md]: replace `title: Kruskal's Algorithm` with `title: "Kruskal's Algorithm"`

File: docs/greedy-algorithms/kruskals-algorithm.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
+---
+id: kruskals-algorithm  
+title: Kruskal's Algorithm  
+sidebar_label: Kruskal's Algorithm
```
Comment (ajay-dhangar) [inline docs/greedy-algorithms/kruskals-algorithm.md]: replace `sidebar_label: Kruskal's Algorithm` with `sidebar_label: "Kruskal's Algorithm"`

File: docs/greedy-algorithms/kruskals-algorithm.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    int V = 4;
+    vector<Edge> edges = {
+        {0, 1, 4}, {0, 2, 3}, {1, 3, 5}, {2, 3, 2}
+    };
+
+    kruskalMST(V, edges);
+
+    return 0;
+}
+
```
Comment (ajay-dhangar) [inline docs/greedy-algorithms/kruskals-algorithm.md:148]: close your code example with ```` ``` ````

File: docs/greedy-algorithms/kruskals-algorithm.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        int V = 4;
+        List<Edge> edges = new ArrayList<>();
+        edges.add(new Edge(0, 1, 4));
+        edges.add(new Edge(0, 2, 3));
+        edges.add(new Edge(1, 3, 5));
+        edges.add(new Edge(2, 3, 2));
+
+        kruskalMST(edges, V);
+    }
+}
```
Comment (ajay-dhangar) [inline docs/greedy-algorithms/kruskals-algorithm.md:229]: Write your code example within ```` ``` ````

for example

````md
```java
// your code here
```
````

File: docs/greedy-algorithms/prims-algorithm.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
+---
+id: prims-algorithm  
+title: Prim's Algorithm  
+sidebar_label: Prim's Algorithm
```
Comment (ajay-dhangar) [inline docs/greedy-algorithms/prims-algorithm.md]: replace `Prim's Algorithm` with `"Prim's Algorithm"`

File: docs/greedy-algorithms/prims-algorithm.md
Comment (ajay-dhangar) [inline docs/greedy-algorithms/prims-algorithm.md:1]: careful read our contribution guideline https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md#d-adding-code-blocks

File: docs/sortings/shell-sort .md
Comment (ajay-dhangar) [inline docs/sortings/shell-sort .md:1]: careful read our contribution guideline https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md#d-adding-code-blocks

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by laxmikandivalasa** [comment]
updated both  the files!!
@ajay-dhangar I'd like to accept for any other changes since this is my first contribution to your repo

**Comment by laxmikandivalasa** [comment]
can you review this pr again!!

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**